azure-functions/src/bindings/durable_orchestration_context.rs

```diff
@@ -19,24 +22,54 @@ use std::{cell::RefCell, collections::HashMap, rc::Rc};
 /// TODO: IMPLEMENT
 #[derive(Debug)]
 pub struct DurableOrchestrationContext {
-    state: Rc<RefCell<OrchestrationState>>,
+    data: DurableOrchestrationContextData,
+    state: Rc<RefCell<ExecutionResult>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct DurableOrchestrationContextData {
+    instance_id: String,
+    is_replaying: bool,
+    parent_instance_id: Option<String>,
+    input: Value,
+    history: Option<Vec<HistoryEvent>>,
 }
 
 impl DurableOrchestrationContext {
     #[doc(hidden)]
     pub fn new(data: TypedData, _metadata: HashMap<String, TypedData>) -> Self {
-        DurableOrchestrationContext {
-            state: Rc::new(RefCell::new(match &data.data {
-                Some(Data::String(s)) => {
-                    from_str(s).expect("failed to parse orchestration context data")
-                }
-                _ => panic!("expected JSON data for orchestration context data"),
-            })),
+        match &data.data {
+            Some(Data::String(s)) => DurableOrchestrationContext {
+                data: from_str(s).expect("failed to parse orchestration context data"),
+                state: Rc::new(RefCell::new(ExecutionResult::default())),
+            },
+            _ => panic!("expected JSON data for orchestration context data"),
         }
     }
 
+    /// Gets the instance ID of the currently executing orchestration.
+    pub fn instance_id(&self) -> &str {
+        &self.data.instance_id
+    }
+
+    /// Gets the parent instance ID of the currently executing sub-orchestration.
+    pub fn parent_instance_id(&self) -> Option<&str> {
+        self.data.parent_instance_id.as_ref().map(|id| &**id)
+    }
+
+    /// Gets a value indicating whether the orchestrator function is currently replaying itself.
+    pub fn is_replaying(&self) -> bool {
+        self.data.is_replaying
+    }
+
+    /// The JSON-serializeable input to pass to the orchestrator function.
+    pub fn input(&self) -> &Value {
+        &self.data.input
+    }
+
     #[doc(hidden)]
-    pub fn get_state(&self) -> Rc<RefCell<OrchestrationState>> {
+    pub fn execution_result(&self) -> Rc<RefCell<ExecutionResult>> {
```

Comment:
We'll need to also fix the code generation to call this method as right now it'll still generate a call to `state`.  Give me a sec and I'll link you to the line.

Reply:
The code is here: https://github.com/peterhuene/azure-functions-rs/blob/durable-functions/azure-functions-codegen/src/func/invoker.rs#L88.

We can probably remove references to "state" and change to "execution_result".